app/code/Magento/MediaGallerySynchronization/Model/Filesystem/FileInfo.php

```diff
@@ -0,0 +1,251 @@
+
+    /**
+     * @var string
+     */
+    private $filename;
+
+    /**
+     * @var string
+     */
+    private $extension;
+
+    /**
+     * @var $basename
+     */
+    private $basename;
+
+    /**
+     * @var string
+     */
+    private $pathname;
+
+    /**
+     * @var int
+     */
+    private $inode;
+
+    /**
+     * @var int
+     */
+    private $size;
+
+    /**
+     * @var int
+     */
+    private $owner;
+
+    /**
+     * @var int
+     */
+    private $group;
+
+    /**
+     * @var int
+     */
+    private $aTime;
+
+    /**
+     * @var int
+     */
+    private $mTime;
+
+    /**
+     * @var int
+     */
+    private $cTime;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var false|string
+     */
+    private $realPath;
+
+    /**
+     * FileInfo constructor.
+     * @param string $file_name
+     * @param string $path
+     * @param string $filename
+     * @param string $extension
+     * @param string $basename
+     * @param string $pathname
+     * @param int $inode
+     * @param int $size
+     * @param int $owner
+     * @param int $group
+     * @param int $aTime
+     * @param int $mTime
+     * @param int $cTime
+     * @param string $type
+     * @param false|string $realPath
+     * @SuppressWarnings(PHPMD.ExcessiveParameterList)
+     */
+    public function __construct(
+        string $file_name,
+        string $path,
+        string $filename,
+        string $extension,
+        string $basename,
+        string $pathname,
+        int $inode,
+        int $size,
+        int $owner,
+        int $group,
+        int $aTime,
+        int $mTime,
+        int $cTime,
+        string $type,
+        $realPath
+    ) {
+        parent::__construct($file_name);
+        $this->path = $path;
+        $this->filename = $filename;
+        $this->extension = $extension;
+        $this->basename = $basename;
+        $this->pathname = $pathname;
+        $this->inode = $inode;
+        $this->size = $size;
+        $this->owner = $owner;
+        $this->group = $group;
+        $this->aTime = $aTime;
+        $this->mTime = $mTime;
+        $this->cTime = $cTime;
+        $this->type = $type;
+        $this->realPath = $realPath;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getPath(): string
+    {
+        return $this->path;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getFilename(): string
+    {
+        return $this->filename;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getExtension(): string
+    {
+        return $this->extension;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getBasename($suffix = null): string
+    {
+        return $this->basename;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getPathname(): string
+    {
+        return $this->pathname;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getInode(): int
```

Comment:
I don't think `Inode` method is used, please keep only properties/methods that are required